### DISCUSSION TITLE
Customize metadata by deleting the artist

### DISCUSSION THREAD

**Discussion by DroidU** [savonet/liquidsoap#3009] [User Support]
I created a script based on the page https://www.liquidsoap.info/doc-dev/metadata.html. The source is an Ogg/Vorbis stream on the Icecast2 server, and the destination is an mp3 format stream also on the Icecast2 server.

```
def append_title(m) =
# Replace title metadata
[("title","my station")]
end

# Apply metadata.map to s using append_title
s = metadata.map(append_title, s)
```

The script works fine until the "artist" in the ogg stream does not get a value.
When artist also gets a value, the target stream's metadata changes to "Artist - my station".
I also deleted the artist with the following modification:

```
def append_title(m) =
# Replace title and artist metadata
[("title","my station"),("artist","")]
end

# Apply metadata.map to s using append_title
s = metadata.map(append_title, s)
```

I was only half-successful with this, because from now on the metadata will always be "- my station", so the first character will be "-".
How can I delete the artist variable so that it is empty in such a way that the starting "-" character disappears in the target stream?

**Comment by vitoyucepi** [comment]
Hi @DroidU,
You could try setting [`metadata.map`](https://www.liquidsoap.info/doc-dev/reference.html#metadata.map) named parameter `update` to `false`.

I think this will do the trick.
```ruby
s = metadata.map(update=false, fun(_) -> [("title", "my station")], s)
```

In this code sample, I use the anonymous function.
The function argument is unused and replaced by `_`.
This function will always return the same value.

Also, you can edit metadata in [`output.icecast`](https://www.liquidsoap.info/doc-dev/reference.html#output.icecast).
Check `icy_metadata` and `icy_song`.

**Reply by DroidU** [reply]
Thanks,
Working perfectly. It don't even need "def append_title(m)".
